test/README.md

```diff
@@ -2,91 +2,129 @@
-| `[E2E_]IOT_HUB_CONNECTION_STRING` | ✔ | Hub-scoped IoT Hub connection string that can be used to get/add/remove devices, deploy edge configurations, and get/update module twins. |
-| `[E2E_]EVENT_HUB_ENDPOINT` | ✔ | Connection string used to connect to the Event Hub-compatible endpoint of your IoT Hub, to listen for D2C events sent by devices or modules. |
-| `[E2E_]REGISTRIES__{n}__PASSWORD` || Password associated with a container registry entry in the `registries` array of `context.json`. `{n}` is the number corresponding to the (zero-based) array entry. For example, if you specified a single container registry in the `registries` array, the corresponding parameter would be `[E2E_]REGISTRIES__0__PASSWORD`. |
-| `[E2E_]ROOT_CA_PASSWORD` || The password associated with the root certificate specified in `rootCaCertificatePath`. |
-| `[E2E_]BLOB_STORE_SAS` || The sas token used to upload module logs and support bundle in the tests. |
-
-_Note: the definitive source for information about test parameters is `test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs`._
-
-### Running the tests
-
-With the test parameters and secrets in place, you can run all the end-to-end tests from the command line
-
-_In Windows Command Prompt (Admin),_
-
-```cmd
-cd {repo root}
-dotnet test test/Microsoft.Azure.Devices.Edge.Test
-```
-_For Linux,_
+2. `test/Microsoft.Azure.Devices.Edge.Test.Common/`: The helper library tests use to do interesting things--like deploy modules or wait for a message to arrive in IoT hub--lives here.
+3. `test/modules`: All custom modules used by the tests live here. The common pattern for an end-to-end test is to deploy some modules that run through a scenario (i.e., the thing you actually want to test, like sending messages between modules), then wait for them to report the results.
+
+## One time setup
+To run the end-to-end tests, we will be building the required binaries from the code, build container images and push them to a local container registry. The tests will install the IoT Edge runtime from the binaries on your machine and run the containers using your local registry. 
+
+To prepare, we will need to setup your local machine and cloud resources. The setup steps are detailed [here](./doc/one-time-setup.md)
+
+## Build
+#### Building the binaries
+From the top folder of the codebase ( i.e., the iotedge folder), run the following(*Note: Some of these steps will require to be run as sudo*)
+~~~ sh
+./scripts/linux/buildBranch.sh
+
+# setup environment variables
+export PACKAGE_OS="ubuntu18.04"
+export PACKAGE_ARCH="amd64"
+
+# update sub modules [DOUBLE CHECK THIS]
+git submodule update --init --recursive
+
+# Create the installable packages. It will typically be located
+# in the ./edgelet/target/release/ and the ./edgelet/target/hsm/ folders 
+# and named something like aziot-edge*amd64.deb and libiothsm-std*.deb 
```

Comment:
This is the 1.2 naming convention.  If this doc is specifically for 1.1, then it should be 'iotedge-1.1.x*amd64.deb'